influence_benchmark/experiments/vec_experiment.py

```diff
@@ -0,0 +1,138 @@
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--env_name", type=str, default="smoking")
+    parser.add_argument("--env_backend_model", type=str, default="meta-llama/Meta-Llama-3-8B-Instruct")
+    parser.add_argument("--agent_backend_model", type=str, default="meta-llama/Meta-Llama-3-8B-Instruct")
+    parser.add_argument("--max_turns", type=int, default=5)
+    parser.add_argument("--num_envs", type=int, default=10)
+    parser.add_argument("--num_episodes", type=int, default=1)
+    parser.add_argument("--device", type=str, default="cuda:5")
+    parser.add_argument("--output_file", type=str, default="data/vec_env_test/results.jsonl")
+    return parser.parse_args()
+
+
+def create_vec_env(args) -> VecEnv:
+    env_configs = []
+    for _ in range(args.num_envs):
+        env_config = {
+            "env_name": args.env_name,
+            "env_backend_model": args.env_backend_model,
+            "max_turns": args.max_turns,
+            "print": False,
+            "device": args.device,
+            "vectorized": True,
+        }
+        env_configs.append(env_config)
+
+    return VecEnv(
+        env_configs=env_configs,
+        PM_backend_model=args.env_backend_model,
+        TM_backend_model=args.env_backend_model,
+        char_backend_model=args.env_backend_model,
+        device=args.device,
+    )
+
+
+def create_agent(args):
+    if args.agent_backend_model in ["gpt-4o", "gpt-4-turbo", "gpt-3.5-turbo"]:
+        return GPTAgent(args.env_name, model_name=args.agent_backend_model)
+    else:
+        return HFAgent(args.env_name, model_name=args.agent_backend_model, device=args.device)
+
+
+def run_episode(vec_env: VecEnv, agent, args) -> List[Dict]:
+    observations = vec_env.reset()
+    done = [False] * args.num_envs
+    episode_data = [[] for _ in range(args.num_envs)]
+
+    for turn in range(args.max_turns):
+        if all(done):
+            break
+        active_observations = [obs for obs, d in zip(observations, done) if not d]
+        actions = agent.get_action_vec(active_observations)
+        padded_actions = []
+        action_index = 0
+        for d in done:
+            if d:
+                padded_actions.append(None)
+            else:
+                padded_actions.append(actions[action_index])
+                action_index += 1
+
+        next_states, done_now = vec_env.step_vec(padded_actions)
+        observations = vec_env.get_observation_vec()
+
+        for i, (state, is_done) in enumerate(zip(next_states, done)):
+            if not is_done:
+                episode_data[i].append(
+                    {
+                        "turn": turn + 1,
+                        "history": state.history,
+                        "preferences": state.preferences,
+                        "transition_probs": state.transition_probs,
+                    }
+                )
+        done = done_now
+    return episode_data
+
+
+@profile()
```

Comment:
Probably want to comment this out when not in use – profiling slows the code down somewhat